File: 2025-09-11 - ListaEncadeada/Exemplo_Java/ListaEncadeadaInterface.java
// ------------------------------------------------------------------------
// Interface da Lista Encadeada
// Todos os métodos retornam booleano (exceto imprimirLista e tamanho)
// ------------------------------------------------------------------------
public interface ListaEncadeadaInterface {
   boolean inserirNoInicio(String codigoDisciplina, String nomeDisciplina, int cargaHoraria);
   boolean inserirNoFim(String codigoDisciplina, String nomeDisciplina, int cargaHoraria);
   boolean inserirNaPosicao(int posicao, String codigoDisciplina, String nomeDisciplina, int cargaHoraria);
   boolean removerPorCodigo(String codigoDisciplina);
   boolean buscarPorCodigo(String codigoDisciplina);
   boolean atualizarDisciplina(String codigoDisciplina, String novoNome, int novaCargaHoraria);
   void imprimirLista();
   int tamanho();
   boolean estaVazia();
}

File: 2025-09-11 - Lista Encadeada/Exemplo_C#/ListaEncadeadaInterface.cs
// ------------------------------------------------------------------------
// Interface da Lista Encadeada
// Todos os métodos retornam bool (exceto imprimirLista e tamanho)
// ------------------------------------------------------------------------
public interface ListaEncadeadaInterface
{
   bool inserirNoInicio(string codigoDisciplina, string nomeDisciplina, int cargaHoraria);
   bool inserirNoFim(string codigoDisciplina, string nomeDisciplina, int cargaHoraria);
   bool inserirNaPosicao(int posicao, string codigoDisciplina, string nomeDisciplina, int cargaHoraria);
   bool removerPorCodigo(string codigoDisciplina);
   bool buscarPorCodigo(string codigoDisciplina);
   bool atualizarDisciplina(string codigoDisciplina, string novoNome, int novaCargaHoraria);
   void imprimirLista();
   int tamanho();
   bool estaVazia();
}
